plugin/pkg/admission/limitranger/admission.go

```diff
@@ -170,7 +170,7 @@ func (l *LimitRanger) GetLimitRanges(a admission.Attributes) ([]*corev1.LimitRan
 			// Fixed: #22422
 			// use singleflight to alleviate simultaneous calls to
 			lruItemObj, err, _ = l.group.Do(a.GetNamespace(), func() (interface{}, error) {
-				liveList, err := l.client.CoreV1().LimitRanges(a.GetNamespace()).List(context.TODO(), metav1.ListOptions{})
+				liveList, err := l.client.CoreV1().LimitRanges(a.GetNamespace()).List(context.TODO(), metav1.ListOptions{ResourceVersion: "0"})
```

Comment:
This change will break this scenario:

1. global resourcequota configuration defaulting to allow no usage of a particular resource (e.g. system-critical pods or some storage class or something)
2. namespace initialization process that rapidly in order creates a) ResourceQuota object allowing use of the globally-restricted resource then b) an object that uses that resource (like a privileged pod or PersistentVolumeClaim or something)

If we're going to make the "live" lookup work from cache, we should just delete the entire `if len(items) == 0 {` block because there's no real difference between the client-held informer-fed cache and the server-held informer-fed cache used with `ResourceVersion=0`.

Reply:
I had not (1) realized would break.

I had noticed (2) but am still struggling a bit with it because I can imagine a user depending on the behavior, but it's a stronger guarantee than we typically provide across the API and removing this to get us the benefits of ConsistentReadsFromCache.

Maybe another way to look at it is: If ConsistentReadsFromCache breaks this use case? Do we expect it to break similar use cases in the ecosystem?

Reply:
ConsistentReadsFromCache can add a maximum of 100ms for resources that don't get any updates as it batches progress notification requests by 100ms. The high latency is caused only by multiple serial requests to a same resource, which is exactly the quota admission case as it makes a LIST per namespace.

Reply:
LimitRange admission *only* intercepts pods and pvcs, and doesn't have the 5 worker bottleneck quota admission does. I'd be more inclined to live with the ConsistentReadFromCache slowdown there for the moment... breaking `kubectl create -f limit_range_and_pvc.yaml` in a namespace initialization scenario where it's currently working doesn't seem great

Reply:
> I'd be more inclined to live with the ConsistentReadFromCache slowdown there for the moment

Great, works for me.